shelley/chain-and-ledger/executable-spec/src/MetaData.hs

```diff
@@ -0,0 +1,121 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE ViewPatterns       #-}
+
+module MetaData
+  ( MetaDatum (..)
+  , MetaData
+  , MetaDataHash (..)
+  , hashMetaData
+) where
+
+import           Cardano.Binary (DecoderError (..), FromCBOR (fromCBOR), ToCBOR (toCBOR))
+import           Cardano.Crypto.Hash (Hash, hash)
+import           Cardano.Ledger.Shelley.Crypto (Crypto, HASH)
+import           Cardano.Prelude (NoUnexpectedThunks (..), Word64, cborError)
+import           Data.Bifunctor (bimap)
+import           Data.Bitraversable (bitraverse)
+import           Data.ByteString as B
+import           Data.ByteString.Lazy as BL
+import           Data.Map.Strict (Map)
+import qualified Data.Text as T
+import qualified Data.Text.Lazy as TL
+
+import qualified Codec.CBOR.Term as CBOR
+
+import           GHC.Generics
+
+-- | A generic metadatum type.
+data MetaDatum
```

Comment:
Any particular reason not to call this `Data` and have a type/newtype for `MetaDatum`? Given that we expect to use this for more stuff in Goguen, that seems like it would reduce the diffs a bit.

Reply:
The name `Data` is a bit vague, I guess which is why you suggest using a new type/newtype. @dcoutts suggested we not try to use the exact same type for the transaction metadata and the plutus data, but instead just keep a translation.

Reply:
Hm, I thought using the same type was part of the attraction.

(I agree that `Data` is a terrible name, I'm open to alternatives!)